draft-pauly-taps-arch.md

```diff
@@ -321,7 +319,7 @@ The Transport System Implementation Concepts define the set of objects used inte
 
 * Path Racing: Path Racing is the act of attempting to establish, or scheduling attempts to establish, multiple Protocol Stacks that differ based on a selection from the available Paths.
 
-* Endpoint Racing: Endpoint Racing is the act of attempting to establish, or scheduling attempts to establish, multiple Protocol Stacks that differ based on the specific representation of the remote and local endpoints, such as IP addresses resolved from a DNS hostname.
+* Endpoint Racing: Endpoint Racing is the act of attempting to establish, or scheduling attempts to establish, multiple Protocol Stacks that differ based on the specific representation of the RemoteEndpoint and the LocalEndpoint, such as IP addresses resolved from a DNS hostname.
```

Comment:
I don't really like this change either—when we're racing Endpoints (which now have no definition...), they're not using "the" original Remote or Local Endpoint, they're using new sets of derived local and remote Endpoints, which don't necessarily carry the full meaning of what is defined as the Local Endpoint and Remote Endpoint to the application. Specifically, I may have a "Local Endpoint" to hold my local state, and a "Remote Endpoint" to define the hostname, certificate policy, etc, for my peer; but the Endpoints representing what I resolved for my Remote Endpoint are sub-addresses that don't need their own state in the same way, but are still different entities, since I am trying different addresses.

Reply:
I think there are two issues here.

Firstly, should "Endpoint Racing" be a concept, or are we just describing Path Racing in a little more detail here? I suspect the latter.

Second, and independent of whether `Endpoint` exists as a super-type of `RemoteEndpoint` and `LocalEndpoint`, how does the racing relate to endpoints. I guess we pass in a list of RemoteEndpoint objects, each of which is resolved to one or more additional `RemoteEndpoint` objects, then we race connections to those remote endpoint? I'm note sure I understand the comment about "don't necessarily carry the full meaning"?